app/(main)/profile/page.tsx

```diff
@@ -60,8 +74,20 @@ export default function ProfilePage() {
   });
 
   const { data: userPositions, isLoading } = useQuery<UserBet[]>({
-    queryKey: ['getUserBets', address],
-    queryFn: async () => await getUserBets(address),
+    queryKey: ['getUserBets', address, page],
+    queryFn: async () =>
+      await getUserBets({ address, page, itemsPerPage: ITEMS_PER_PAGE }),
+    enabled: !!address,
+  });
+
+  const { data: nextPagePositions } = useQuery<UserBet[]>({
+    queryKey: ['getUserBetsNextPage', address, page],
+    queryFn: async () =>
+      await getUserBets({
+        address,
+        itemsPerPage: ITEMS_PER_PAGE,
+        page: page + 1,
+      }),
```

Comment:
I know we've done it like this in past, but would be more efficient to query how many items does it have?

Like doing a simple query with the max, 999, as see how much items it has. Most of the queries will be bellow that, and we have `totalItems`, in case it's more we can do another query when changing page.

Reply:
we can address this in another PR, I think we need to revaluate how we do pagination on thegraph